src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -1087,13 +1097,13 @@ public String getIamResourceParamName(Field field) {
    * class.
    */
   public String getPathTemplateName(
-      Interface apiInterface, SingleResourceNameConfig resourceNameConfig) {
+      String apiInterfaceSimpleName, SingleResourceNameConfig resourceNameConfig) {
```

Comment:
and SurfaceNamers use InterfaceModel as parameter instead of String whenever possible.